code/modules/buildmode/submodes/delete.dm

```diff
@@ -0,0 +1,62 @@
+/datum/buildmode_mode/delete
+	key = "delete"
+
+/datum/buildmode_mode/delete/show_help(client/c)
+	to_chat(c, "<span class='notice'>***********************************************************\n\
+		Left Mouse Button on anything to delete it. If you break it, you buy it.\n\
+		Right Mouse Button on anything to delete everything of the type. Probably don\'t do this unless you know what you are doing.\n\
+		***********************************************************</span>")
+
+/datum/buildmode_mode/delete/handle_click(client/c, params, object)
+	var/list/pa = params2list(params)
+	var/left_click = pa.Find("left")
+	var/right_click = pa.Find("right")
+
+	if(left_click)
+		if(isturf(object))
+			var/turf/T = object
+			T.ScrapeAway(flags = CHANGETURF_INHERIT_AIR)
+		else if(isatom(object))
+			qdel(object)
+
+	if(right_click)
+		if(check_rights(R_DEBUG|R_SERVER)) //Prevents buildmoded non-admins from breaking everything.
```

Comment:
early return this

Reply:
it retruns on `if(isturf(object))` on the next line. If we return before that it just wont run